### PR TITLE
[MIG] Migrating to Odoo 13.0

### PR DESCRIPTION
**Pull request by macagua**
For install you need resolved the following Python dependencies: 

```
$ pip install geopy==2.2.0
```

For install the website_shops_map addon, you need resolved the following add-on dependencies: 

- [base_multi_image](https://github.com/macagua/server-tools/tree/13.0/base_multi_image) for Odoo 13.0.

### CODE REVIEW

**Comment by macagua** [comment]
@eyekraft @Tatider please, can you check this pull request?

File: __init__.py [hunk 1]
```diff
@@ -1,44 +1,46 @@
 # -*- coding: utf-8 -*-
```
Comment (Tatider) [inline __init__.py]: Here and in all python files: Redundant since Odoo13 uses python3

File: __init__.py [hunk 2]
```diff
@@ -1,44 +1,46 @@
...
-import models
-import controllers
+from . import controllers
+from . import models
+from . import setup
 
 
 def fn_post_init_hook(cr, registry):
-    from openerp import api, SUPERUSER_ID
+    from odoo import api, SUPERUSER_ID
```
Comment (Tatider) [inline __init__.py]: Can be moved to the imports section at the start of the file and combined with the line `from . import models`

File: data/yandex_maps_data.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="0">
```
Comment (Tatider) [inline data/yandex_maps_data.xml]: Tag `data` can be dropped in version 13.0

File: models/eyekraft_shop.py [hunk 4]
```diff
@@ -21,127 +21,148 @@
 def urlplus(url, params):
     return werkzeug.Href(url)(params or None)
 
-# class to load extended description of module
+
 class eyekraft_module_description(models.Model):
+    """
+    Class to load extended description of module.
+    """
     _name = "ir.module.module"
```
Comment (Tatider) [inline models/eyekraft_shop.py]: I would recommened to split each class to separate file - helps with logic analysis and navigation

File: models/eyekraft_shop.py [hunk 5]
```diff
@@ -191,47 +212,50 @@ def _compute_full_address(self):
...
     work_hours_ids = fields.One2many(
         "shop.work.hours",
         "shop_id",
         string="Work hours",
+        help='In this field, you can enter opening hours and enter the data in the window that appears, or select previously entered hours from the list',
+    )
+    rating = fields.Float(
+        string="Rating"
+    )
+    public = fields.Boolean(
```
Comment (Tatider) [inline models/eyekraft_shop.py]: There is a good mixin in Odoo core for that https://www.odoo.com/documentation/13.0/developer/reference/addons/mixins.html#website-visibility

File: models/eyekraft_shop.py [hunk 6]
```diff
@@ -191,47 +212,50 @@ def _compute_full_address(self):
...
+        help='In this field, you can select the services available in the shop in the "Properties" field (services must be pre-configured at "Shop Properties")',
     )
-
     work_hours_ids = fields.One2many(
         "shop.work.hours",
         "shop_id",
         string="Work hours",
+        help='In this field, you can enter opening hours and enter the data in the window that appears, or select previously entered hours from the list',
+    )
+    rating = fields.Float(
```
Comment (Tatider) [inline models/eyekraft_shop.py]: There is a mixin for that in Odoo core: https://www.odoo.com/documentation/13.0/developer/reference/addons/mixins.html#customer-rating

File: models/eyekraft_shop.py [hunk 7]
```diff
@@ -191,47 +212,50 @@ def _compute_full_address(self):
...
-
-    public = fields.Boolean(string="Shop")
-
     foreign_partner = fields.Boolean(help="Flag for partner record linked on module install")
-
     warehouse_ids = fields.One2many(
         "stock.warehouse",
         "shop_id",
-        "Warehouse",
+        string="Warehouse Name",
```
Comment (Tatider) [inline models/eyekraft_shop.py]: Should it be "Warehouses"?

File: models/inherit_stock.py [hunk 8]
```diff
@@ -15,31 +12,61 @@ class stock_warehouse(models.Model):
...
+        if not self.full_address:
+            return 0,0
+            if 'geopy' in sys.modules:
+                # Geocoder using the Yandex Maps API.
+                geolocator = geocoders.Yandex(api_key=YANDEX_API_KEY)
+
+                # https://geopy.readthedocs.io/en/latest/index.html#geopy.geocoders.Yandex.geocode
+                spanish = ["es_ES", "es_AR", "es_BO", "es_CL", "es_CO", "es_CR", "es_DO", "es_EC", "es_GT", "es_MX", "es_PA", "es_PE", "es_PY", "es_UY", "es_VE"]
+                english = ["en_AU", "en_CA", "en_GB", "en_US"]
+                if self.env.context['lang'] == u'es' or self.env.context['lang'] in spanish:
```
Comment (Tatider) [inline models/inherit_stock.py]: I believe it would make sense to create a variable `lang = self.env.context['lang']` and use it instead of taking it from context all the time.

File: models/inherit_stock.py [hunk 9]
```diff
@@ -15,31 +12,61 @@ class stock_warehouse(models.Model):
...
+                elif self.env.context['lang'] == u'uk' or self.env.context['lang'] == u'uk_UA':
+                    language="uk_UA"
+                elif self.env.context['lang'] == u'be' or self.env.context['lang'] == u'be_BY':
+                    language="be_BY"
+                elif self.env.context['lang'] == u'ru' or self.env.context['lang'] == u'ru_RU':
+                    language="ru_RU"
+                else:
+                    language="en_RU"
+
+                for i in xrange(10):
```
Comment (Tatider) [inline models/inherit_stock.py]: What is this loop for? If it is needed, then would be great to have a comment on that.

File: models/inherit_stock.py [hunk 10]
```diff
@@ -49,63 +76,65 @@ def unlink(self):
...
-	if record.shop_id.public:
-	    values['company_type'] = 'company'
-	    values['is_company'] = True
-	    values['category_id'] = [(6,0,[self.env.ref('website_shops_map.partner_shop_category_id').id])]
-	record.shop_id.write(values)
-	return record
-
-    @api.one
+        record = super(stock_warehouse, self).create(vals)
+        values = {}
```
Comment (Tatider) [inline models/inherit_stock.py]: Write doesn't make sense w/o values. So next refactoring would be nice.
```
        if record.shop_id.public:
            values = {
                "company_type": "company",
                "is_company": True,
                "category_id": [
                    (6, 0, [self.env.ref('website_shops_map.partner_shop_category_id').id]),
                ],
            }
            record.shop_id.write(values)
```

File: models/inherit_stock.py [hunk 11]
```diff
@@ -49,63 +76,65 @@ def unlink(self):
...
-		    values['image'] = self.env['res.company'].browse(self.partner_id.id).logo
-	self.shop_id.write(values)
-	return record
-
-
-    @api.multi
+        record = super(stock_warehouse, self).write(vals)
+        values = {'name': self.name}
+        # if warehouse is shop
+        if self.shop_id.public:
```
Comment (Tatider) [inline models/inherit_stock.py]: 1. Having a variable for `self.shop_id` would be nice
2. Write can be called on recordset (more than one record), in such case `self.shop_id.public` will raise singleton error.

File: models/inherit_stock.py [hunk 12]
```diff
@@ -49,63 +76,65 @@ def unlink(self):
...
 
-    @api.multi
     def _set_address(self, field):
         setattr(self, field, getattr(self.partner_id, field))
 
     @api.model
     def erase(self):
-	for rec in self.search([]):
-	    rec.sudo().unlink()
+        for rec in self.search([]):
```
Comment (Tatider) [inline models/inherit_stock.py]: suggestion to replace with :
```
records = self.search(
    [],
)
records.sudo().unlink()
```

File: models/inherit_stock.py [hunk 13]
```diff
@@ -49,63 +76,65 @@ def unlink(self):
...
+            rec.sudo().unlink()
 
     @api.model
     def erase_field(self,model,field):
-	recs = self.env[model].search([('name','=',field)])
-	for rec in recs:
-	    rec.sudo().unlink()
+        recs = self.env[model].search([('name','=',field)])
```
Comment (Tatider) [inline models/inherit_stock.py]: suggestion to replace with :
```
records = self.search(
    [('name', '=', field)],
)
records.sudo().unlink()
```

File: models/inherit_stock.py [hunk 14]
```diff
@@ -49,63 +76,65 @@ def unlink(self):
...
 
     @api.model
     def erase(self):
-	for rec in self.search([]):
-	    rec.sudo().unlink()
+        for rec in self.search([]):
+            rec.sudo().unlink()
 
     @api.model
     def erase_field(self,model,field):
```
Comment (Tatider) [inline models/inherit_stock.py]: missing whitespaces

File: models/inherit_stock.py [hunk 15]
```diff
@@ -1,12 +1,9 @@
 # -*- coding: utf-8 -*-
```
Comment (Tatider) [inline models/inherit_stock.py]: The file name should be the same as the model in it.

File: models/res_config_settings.py [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+    ('ru_RU', 'Russian'),
+    ('tr_TR', 'Turkish'),
+    ('uk_UA', 'Ukrainian'),
+]
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+
```
Comment (Tatider) [inline models/res_config_settings.py]: Too many blank lines

File: models/res_partner.py [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+
+    # Model Fields
+    partner_latitude = fields.Float(string='Geo Latitude', digits=(16, 5))
+    partner_longitude = fields.Float(string='Geo Longitude', digits=(16, 5))
+    date_localization = fields.Date(string='Geolocation Date')
+    image = fields.Binary(string="Partner Image", attachment=True,
```
Comment (Tatider) [inline models/res_partner.py]: If a photo for the shop why it's not in the shop model? Especially then were can be more that 1 shop linked to a partner.

File: models/shop_work_hours.py [hunk 18]
```diff
@@ -35,31 +35,43 @@ def _compute_name(self):
...
             )
 
-    name = fields.Char("Period of Time", compute="_compute_name", store=True)
-    start_time = fields.Float("Since", required=True)
-    end_time = fields.Float("To", required=True)
+    # Model Fields
+    name = fields.Char(
+        string="Period of Time",
+        compute="_compute_name",
+        store=True
```
Comment (Tatider) [inline models/shop_work_hours.py]: Here and below: Trailing comma is missing

File: static/lib/jquery.autocomplete.js [hunk 19]
```diff
@@ -12,34 +12,33 @@
 // Expose plugin as an AMD module if AMD loader is present:
```
Comment (Tatider) [inline static/lib/jquery.autocomplete.js:12]: Is there a new version available of the autocomplete?

File: static/lib/jquery.cookie.js [hunk 20]
```diff
@@ -6,109 +6,112 @@
  * Released under the MIT license
```
Comment (Tatider) [inline static/lib/jquery.cookie.js:6]: Have you checked for newer version?

File: static/src/js/shop_list_editor.js [hunk 21]
```diff
@@ -1,349 +1,476 @@
...
+            this.$("label[for=link-new]").parent().replaceWith(function(index,oldHTML) {
+                return $('<h5 class="list-group-item-heading">').html(oldHTML)
+            })
+            this.$("label[for=link-new]").text(_t("Source API-key"));
+            this.$("label[for=link-external]").parent().replaceWith(function(index,oldHTML) {
+                return $('<h5 class="list-group-item-heading">').html(oldHTML)
+            })
+            this.$("label[for=link-external]").text(_t("Source URL"));
+            var mainDiv = this.$("div[class=list-group]").parent();
+            var divListGroup = $(
```
Comment (Tatider) [inline static/src/js/shop_list_editor.js:250]: Can be defined as a template in xml file and parsed here as via `qweb` from Odoo JS Core

File: models/res_partner.py [hunk 22]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+import logging
+from odoo import fields, models
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+
```
Comment (Tatider) [inline models/res_partner.py:10]: Too many blank lines

File: controllers/main.py [hunk 23]
```diff
@@ -48,23 +67,23 @@ def get_shops_list(self, query=False,  debug=False, lat=False, lng=False, **kwar
...
+        tags = request.params.get('tags','')
+        tags = tags.split(',') if tags else False
+        shop_ids = []
+        if not isIds:
+            # if not ids list passes get recordset of all shops ('public' flag in true)
+            # checks if linked warehouse record is not archieved ('active' flag is true)
+            # and filter by selected tags
+            domain = [('public', '=', True),('warehouse_ids.active','=',True)]
+            if tags:
+                domain.append('|') if len(tags) > 1 else None
```
Comment (Tatider) [inline controllers/main.py:79]: It recommended to use expressions to work with domains
https://github.com/odoo/odoo/blob/13.0/odoo/osv/expression.py

File: controllers/main.py [hunk 24]
```diff
@@ -133,4 +152,4 @@ def get_xml_templates(self, debug=False, **kwargs):
             result += ("\n").join(templates.arch_base.split("\n")[1:])
         result += "</templates>"
 
-        return result
\ No newline at end of file
+        return result
```
Comment (Tatider) [inline controllers/main.py:154]: `ajax.loadXM` can load XML by file names. Why bother with a route?

File: __init__.py [hunk 25]
```diff
@@ -1,44 +1,46 @@
...
-	warehouse = model_obj.create(values)
-	warehouse.public = True
-	warehouse.shop_id.foreign_partner = True
-	partner_to_delete = warehouse.shop_id.partner_id
-	warehouse.shop_id.partner_id = partner
-	partner_to_delete.sudo().unlink()
-	i += 1
\ No newline at end of file
+        warehouse = model_obj.create(values)
+        warehouse.public = True
```
Comment (Tatider) [inline __init__.py]: It is a risk make something public automatically...

**Review by Tatider — COMMENTED** [review]
Code can be improved according to pep8, then it would be easier to go through it.

File: models/res_partner.py (hunk 22 above)
Comment (macagua) [reply models/res_partner.py:10]: Done with commit [f2a28e0](https://github.com/macagua/website_shops_map/commit/f2a28e0d32e9dc6a2ac752762b4c7469f5d3dc10)

File: __init__.py (hunk 1 above)
Comment (macagua) [reply __init__.py]: Done with commits [d1daa82](https://github.com/macagua/website_shops_map/commit/d1daa820cea57cd958d086662d10cc09ec0fd844), [248dc6e](https://github.com/macagua/website_shops_map/commit/248dc6eb5d1770dbdf215547547816bccbe6acc1) and [a45d4f5](https://github.com/macagua/website_shops_map/commit/a45d4f511819c2f0e2c4c056e5fd315f0a1e9703)

File: __init__.py (hunk 2 above)
Comment (macagua) [reply __init__.py]: Done with commit [6bde8b6](https://github.com/macagua/website_shops_map/commit/6bde8b6a23e4c4f10c74f1f6e66507c5a13fb0b0)

File: models/eyekraft_shop.py (hunk 4 above)
Comment (macagua) [reply models/eyekraft_shop.py]: Done with commit [67f9590](https://github.com/macagua/website_shops_map/commit/67f95908e00cce4be663987082c23bf9b57db9bc)

File: data/yandex_maps_data.xml (hunk 3 above)
Comment (macagua) [reply data/yandex_maps_data.xml]: Donw with commit [65211c4](https://github.com/macagua/website_shops_map/commit/65211c49f77aa158ae12679b36b5339e7d8c3077)

File: models/res_config_settings.py (hunk 16 above)
Comment (macagua) [reply models/res_config_settings.py]: Done with commit [67f9590](https://github.com/macagua/website_shops_map/commit/67f95908e00cce4be663987082c23bf9b57db9bc)

File: models/inherit_stock.py (hunk 9 above)
Comment (macagua) [reply models/inherit_stock.py:46]: @Tatider If I understand: The structure is always IF then ELIF and finally ELSE (which is executed in case none of the previous IF/ELIF is fulfilled), what do you think?

File: models/inherit_stock.py (hunk 8 above)
Comment (macagua) [reply models/inherit_stock.py]: Done with commits [09fb3b8](https://github.com/macagua/website_shops_map/commit/09fb3b8ae5128a2418b8169e8440240fb0ebc706) and [e5c51d5](https://github.com/macagua/website_shops_map/commit/e5c51d5a3fbe37c8716cb64c5314811946ca79a7)

File: models/inherit_stock.py (hunk 15 above)
Comment (macagua) [reply models/inherit_stock.py]: Done with commits [c7c3511](https://github.com/macagua/website_shops_map/commit/c7c351154f75935ac369d3ac3e5f243894311142) and [ddc16d5](https://github.com/macagua/website_shops_map/commit/ddc16d5d3bafa4088aac4593ca78fe1a15a53cda)

File: models/shop_work_hours.py (hunk 18 above)
Comment (macagua) [reply models/shop_work_hours.py]: Done with commit [0fdb65d](https://github.com/macagua/website_shops_map/commit/0fdb65d8228ea5c2c66cb9cc341f38f5f2df6353)

File: models/eyekraft_shop.py (hunk 7 above)
Comment (macagua) [reply models/eyekraft_shop.py]: @Tatider @eyekraft I changed this to fixed WARNING odoo.addons.base.models.ir_model: Two fields (warehouse_ids, name) of stock.warehouse() have the same label: Warehouse, with commit [e809d3d](https://github.com/macagua/website_shops_map/commit/e809d3d1e10ef06da23c60071780f7d35888eef6)

File: static/lib/jquery.autocomplete.js (hunk 19 above)
Comment (macagua) [reply static/lib/jquery.autocomplete.js:12]: @Tatider We have the [v1.2.27](https://github.com/devbridge/jQuery-Autocomplete/releases/tag/v1.2.27) release on 17 Nov 2016 for Ajax Autocomplete for jQuery Plugin, his last release was [v1.4.11](https://github.com/devbridge/jQuery-Autocomplete/releases/tag/v1.4.11) on 1 May 2020

File: static/lib/jquery.cookie.js (hunk 20 above)
Comment (macagua) [reply static/lib/jquery.cookie.js:6]: We have the [v1.4.1](https://github.com/carhartl/jquery-cookie/releases/tag/v1.4.1) release on 3 Apr 2017 for jQuery Cookie Plugin, his last release was 1.4.1 because reading the README file said "This project was moved to https://github.com/js-cookie/js-cookie... View documentation for the latest backwards compatible release (1.5.1)". then checking the [README for js-cookie v1.5.1](https://github.com/js-cookie/js-cookie/tree/v1.5.1) said "Migrating from jquery-cookie", is a guite to migrate from jquery-cookie to JavaScript Cookie

Comment (macagua) [reply static/lib/jquery.cookie.js:6]: Done with commits [f3cf488](https://github.com/macagua/website_shops_map/commit/f3cf488a9db6727507ef4b6ef6e44137583d6017) and [3c90f59](https://github.com/macagua/website_shops_map/commit/3c90f59e7957d1f81fc4a2af4a938d9707e94eb6)
